packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -564,6 +568,91 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     }
   }
 
+  async signIn(
```

Comment:
If it makes sense, please use JSON input here, I moved other functions to use it where it makes sense